Sources/FluentKit/Query/Builder/QueryBuilder+Filter.swift

```diff
@@ -43,6 +43,28 @@ extension QueryBuilder {
     {
         self.filter(Model.path(for: lhsField), method, Model.path(for: rhsField))
     }
+    
+    @discardableResult
+    public func filter<LeftJoined, LeftField, RightJoined, RightField>(
+        _ leftJoined: LeftJoined.Type,
+        _ leftField: KeyPath<LeftJoined, LeftField>,
+        _ method: DatabaseQuery.Filter.Method,
+        _ rightJoined: RightJoined.Type,
```

Comment:
Could we add a default value to the `leftJoined` and `rightJoined` parameters so we can use this syntax to call the method?

```swift
.filter(\FooAlias.$number, .equal, \BarAlias.$number)
```